pkg/api/v20191231preview/openshiftcluster.go

```diff
@@ -81,6 +81,9 @@ const (
 
 // ClusterProfile represents a cluster profile.
 type ClusterProfile struct {
+	// The pull secret for the cluster (immutable).
+	PullSecret string `json:"pullSecret,omitempty"`
```

Comment:
@amanohar proposed API change.  To begin with, PullSecret is write-only and immutable.  Hopefully we can make it write-only + mutable later.